src/components/compute-type-picker.tsx

```diff
@@ -9,43 +9,28 @@ export type ComputeTypePickerProps = {
   name: string;
   id: string;
   environment: string;
+  environmentList: Scheduler.IRuntimeEnvironment[];
   onChange: (event: SelectChangeEvent<string>) => void;
   initialValue: string;
 };
 
-export function computeTypesForEnvironment(
-  environment: string
-): string[] | null {
-  // Retrieve the environment data from session storage.
-  const environmentsData = sessionStorage.getItem('environments');
-  if (environmentsData === null) {
-    return null;
-  }
-
-  const environments = JSON.parse(
-    environmentsData
-  ) as Array<Scheduler.IRuntimeEnvironment>;
-  const environmentObj = environments.find(env => env.name === environment);
-  if (!environmentObj || !environmentObj['output_formats']) {
-    return null;
-  }
-
-  return environmentObj['compute_types'];
-}
-
 export function ComputeTypePicker(
   props: ComputeTypePickerProps
 ): JSX.Element | null {
-  const computeTypes = useMemo(
-    () => computeTypesForEnvironment(props.environment),
-    [props.environment]
+  const environmentObj = props.environmentList.find(
+    env => env.name === props.environment
   );
-  if (computeTypes === null) {
+  if (!environmentObj || !environmentObj['compute_types']) {
     return null;
   }
 
+  const computeTypes = environmentObj['compute_types'] as string[];
+
   const labelId = `${props.id}-label`;
 
+  // If no initial value was provided, default to the first value being selected.
+  const initialValue = props.initialValue || computeTypes[0];
+
```

Comment:
I just realized how poorly named `props.initialValue` is. It's not an initial value, it actually does set the value for this field. So, there's a few things wrong here.

1) Rename `props.initialValue` to `props.value`.

2) The default value of a field should never be set by the field itself, since there's no way for the parent form component to read it and actually submit it to the backend. This is essentially just for rendering purposes only and is actually quite misleading, since `props.model.computeType` (the value actually sent to the backend) can still be nullish.

Thus, the default value needs to be lifted up and immediately set by the form after `environment` gets selected. In `create-job.tsx`:

```
  const handleSelectChange = (event: SelectChangeEvent<string>) => {
    const target = event.target as HTMLInputElement;

    // if setting environment, default to first compute type
    if (target.name === 'environment') {
      const envObj = environmentList.find(env => env.name === target.value)
      props.handleModelChange({ ...props.model, environment: target.value, computeType: envObj.compute_types?.[0] })
    else {
      // otherwise just set the model
      props.handleModelChange({ ...props.model, [target.name]: target.value });
    }
  };
```

Reply:
Also, keep in mind that this control flow is necessary because you should only be invoking `handleModelChange` once per render. This is because we're not use the callback syntax of `setState`, meaning that:

```
props.handleModelChange({ ...props.model, a: "asdf" })
props.handleModelChange({ ...props.model, b: "asdf" })
```

actually only sets property `b`. This is because within the lifetime of this function, `props.model` refers to the same object.